skipruntime-ts/native/src/Runtime.sk

```diff
@@ -239,16 +247,35 @@ fun buildResourcesGraph(
     dirname,
     Array[],
   );
-  _ = resourcesHdl.map(
-    SKStore.SID::keyType,
+  idbyresourceHdl = resourcesHdl.map(
+    ResourceDef::keyType,
+    SKStore.StringFile::type,
+    context,
+    dirname.sub("idbyresource"),
+    (_c, writer, key, it) ~> {
+      writer.set(it.first, SKStore.StringFile(key.value));
+    },
+  );
+  availablesHdl = idbyresourceHdl.map(
+    ResourceDef::keyType,
+    SKStore.IntFile::type,
+    context,
+    dirname.sub("availables"),
+    (_c, writer, key, _it) ~> {
+      // map to ignore array changes
+      // to prevent recompute on each add/remove session
+      writer.set(key, SKStore.IntFile(1))
+    },
+  );
+  _ = availablesHdl.map(
```

Comment:
Why not directly map on `resourcesHdl` and use `it.first`?

Reply:
Because each source will compute its own graph, it's allow to have have one source force a resource reactive graph.